src/pages/SignIn.tsx

```diff
@@ -13,278 +14,292 @@ import Stack from '@mui/material/Stack';
+          setEmailError(false);
+          setEmailErrorMessage('');
         }
-    };
 
-    const [password, setPassword] = useState('');
-    const [passwordError, setPasswordError] = useState(false);
-    const handleChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setPassword(e.target.value);
-    };
-    const handleBlurPassword = () => {
-        if (!validatePassword(password)) {
-            setPasswordError(true);
+        if (!password.value || !passwordRegex.test(password.value) ) {
+          setPasswordError(true);
+          setPasswordErrorMessage('Password must be at least 8 characters long and include an uppercase letter, a lowercase letter, a number, and a special character.');
+          isValid = false;
         } else {
-            setPasswordError(false);
+          setPasswordError(false);
+          setPasswordErrorMessage('');
         }
-    };
-
-    const [open, setOpen] = useState(false);
-
-    // Call useSignUpNavigation inside the component to follow hook rules
-    const navigateToSignUp = useSignUpNavigation();
 
-    const handleClickOpen = () => {
-        setOpen(true);
+        return isValid;
     };
 
-    const handleClose = () => {
-        setOpen(false);
-    };
-
-    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-        if (emailError || passwordError) {
-            event.preventDefault();
-            return;
+// server site validation for email and password
+    const attemptSignIn: Promise<boolean> = async (email, password) => {
+        try {
+            const response = await penguinApi.post('http://localhost:8080/api/v1/user/login',
+                {
+                    'email': email.value,
+                    'password': password.value
+                }
+            );
+            sessionStorage.setItem('token', response.data.token);
+            return response.status === 200;
+        } catch (error) {
+            console.error('Error validating user:', error);
+            return false;
         }
-        const data = new FormData(event.currentTarget);
-        console.log({
-            email: data.get('email'),
-            password: data.get('password'),
-        });
     };
 
-    const validateEmail = (email: string) => {
-        return /\S+@\S+\.\S+/.test(email);
-    };
+    const onSignInClick = () => {
+        const email = document.getElementById('email') as HTMLInputElement;
+        const password = document.getElementById('password') as HTMLInputElement;
 
-    const validatePassword = (password: string) => {
-        return password.length > 5;
+        if (validateInputs(email, password)) {
+            attemptSignIn(email, password)
+                .then((isSuccess: boolean) => {
+                    if (isSuccess) {
+                        console.log("Success");
+                        navigateHome();
+                    } else {
+                        console.error("Bad Login credentials");
```

Comment:
If the sign-in attempt is unsuccessful, we should communicate this to the user somehow instead of just logging it to console.